pkg/controller/subscription/subscription_controller.go

```diff
@@ -292,8 +294,10 @@ func (r *ReconcileSubscription) doReconcile(instance *appv1alpha1.Subscription)
 			Namespace: instance.Namespace,
 		}
 
-		err = r.hubclient.Get(context.TODO(), subcfgkey, subitem.SubscriptionConfigMap)
-		if err != nil {
+		errLocal := r.Client.Get(context.TODO(), subcfgkey, subitem.SubscriptionConfigMap)
+		errRemote := r.hubclient.Get(context.TODO(), subcfgkey, subitem.SubscriptionConfigMap)
+
+		if errRemote != nil && errLocal != nil {
```

Comment:
here the idea is checking if the reference `configmap` exists on hub or local. So, if both of them says no, then we really have a problem. Otherwise, we are fine.

Reply:
As the `obj := obj`, the `obj` is a pointer within the `for` loop, if we don't do this, then the `obj` pointer might cause us some issue.

The sonar cloud pointed out this one. 

https://github.com/golang/go/issues/20733